1187 migrate collections and spaces as v2 metadata.

tcnichol opened the pull request:
This is a work in progress for now.

To test this, you will first need to have an admin user in v2, and you will have to add the metadata definitions. I have added a new 'Collection' metadata definition. Right now this just has a field for the collection_name and collection_id, along with fields for parents. To handle nested collections beyond 1 level, or to handle datasets that are in more than one collection, the definition will need to change.

longshuicy commented on scripts/migration/migrate.py:
```diff
@@ -117,6 +117,110 @@ def add_v1_space_members_to_v2_group(space, group_id, headers):
+    for collection in user_collections:
+        collection_id = collection["id"]
+        if collection["name"] == "subchild":
+            print("HERE")
```
Is this print needed?

longshuicy commented on scripts/migration/migrate.py:
```diff
@@ -117,6 +117,110 @@ def add_v1_space_members_to_v2_group(space, group_id, headers):
+    response = requests.get(endpoint, headers=headers)
+    self = response.json()
+    if "id" not in self:
+        print("HERE")
```
same Is this print needed?

tcnichol replied:
will remove, i was using this print in debugging

longshuicy commented on scripts/migration/migrate.py:
```diff
@@ -117,6 +117,110 @@ def add_v1_space_members_to_v2_group(space, group_id, headers):
+    for collection in all_collections:
+        collection_name = collection["name"]
+        if collection_name == "ROOT C" or collection_name == "ROOT D":
+            print("C OR D")
```
What does Root C or Root D mean?

tcnichol replied:
oops. i was checking what happened some specific collections from v1, removed.

tcnichol commented:
I think I've addressed all comments and everything is working. Got rid of all the print statements I had accidentally checked in for debugging and did some better logging for space and collection.

For posting the space metadata for a dataset, I added a try/except so if we're using v1 without my fix to return the space author, we will post space metadata based on the fields that we have.
